### PR TITLE
TF-2758 Email event description in html

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2758 

### Before
Screenshot 2024-04-01 at 16 12 43
Screenshot 2024-04-01 at 16 13 01
### After
Screenshot 2024-04-01 at 16 13 21
Screenshot 2024-04-12 at 11 25 40

### CODE REVIEW

**Comment by chibenwa** [comment]
image

That's HTML in the event.

image

And that's HTML in the JMAP call too.

Please directly render HTML (after sanitizing) rather than passing through a plain text escaped representation.

**Review by chibenwa — CHANGES_REQUESTED** [review]
CF earlier comment

File: core/lib/presentation/extensions/string_extension.dart [hunk 1]
```diff
@@ -46,4 +46,12 @@ extension StringExtension on String {
...
       .toString();
   }
+
+
+  String get decodeHtml {
+    return replaceAll('&lt;', '<')
+      .replaceAll('&gt;', '>')
+      .replaceAll('&quot;', '"')
+      .replaceAll('&amp;', '&');
+  }
```
Comment (hoangdat) [inline core/lib/presentation/extensions/string_extension.dart:56]: did we have any HTML transformer?

Comment (hoangdat) [inline core/lib/presentation/extensions/string_extension.dart:56]: did we have any HTML transformer?

**Comment by tddang-linagora** [comment]
> image
> 
> That's HTML in the event.
> 
> image
> 
> And that's HTML in the JMAP call too.
> 
> Please directly render HTML (after sanitizing) rather than passing through a plain text escaped representation.

Done

File: core/lib/presentation/extensions/string_extension.dart (hunk 1 above)
Comment (hoangdat) [reply core/lib/presentation/extensions/string_extension.dart:56]: how about using `TransformConfiguration` to handle html? WDYT @dab246

**Review by dab246 — APPROVED** [review]
lgtm

**Review by hoangdat — APPROVED** [review]

**Pull request merged**